x-pack/plugins/endpoint/server/index_pattern.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { SavedObjectsClientContract, Logger, LoggerFactory } from 'kibana/server';
+import { IndexPatternService } from '../../ingest_manager/server';
+
+const endpointPackageName = 'endpoint';
+
+export interface IndexPatternRetriever {
+  get(client: SavedObjectsClientContract, datasetPath: string): Promise<string>;
+}
+
+export class IngestIndexPatternRetriever implements IndexPatternRetriever {
+  private readonly log: Logger;
+  constructor(private readonly service: IndexPatternService, loggerFactory: LoggerFactory) {
+    this.log = loggerFactory.get('index-pattern-retriever');
+  }
+
+  async get(client: SavedObjectsClientContract, datasetPath: string, version?: string) {
+    const pattern = await this.service.get(client, endpointPackageName, datasetPath);
+
+    if (!pattern) {
+      this.log.warn(`Failed to retrieve index pattern from ingest manager dataset: ${datasetPath}`);
+      throw new Error(`Unable to retrieve the index pattern for dataset: ${datasetPath}`);
```

Comment:
This will still result in a error that gets logged, is that not this case? Why log warn and throw error? Also why are the messages different?

Reply:
I think the error message will only get logged if there's an explicit log line in the handler that calls this class. For example it will get logged here:
https://github.com/jonathan-buttner/kibana/blob/endpoint-ingest-index-pattern/x-pack/plugins/endpoint/server/routes/index_pattern.ts#L28

But I don't think it'll get logged here: https://github.com/jonathan-buttner/kibana/blob/endpoint-ingest-index-pattern/x-pack/plugins/endpoint/server/routes/metadata/index.ts#L74

The kibana logs would just show that a 500 occurred but in the console I don't think it'll print the error message.

Yeah I'll change it so the messages are the same.